src/tiledb/cloud/bioimg/exportation.py

```diff
@@ -64,36 +76,66 @@ def build_io_uris_exportation(
-                    # Folder for exploration
-                    contents = vfs.ls(uri)
-                    yield from iter_paths(contents)
-                elif tiledb.object_type(uri) == "group":
-                    # For exportation we require the source path to be a tiledb group
-                    yield uri, create_output_path(uri, output_dir)
-
-        if len(source) == 0:
-            raise ValueError("The source files cannot be empty")
-        return tuple(iter_paths(source))
+        def create_output_path(input_file: str, output: str) -> str:
+            # Check if output is dir
+            if output.endswith("/"):
+                filename = os.path.splitext(os.path.basename(input_file))[0]
+                output_filename = (
+                    filename + f".{output_ext}" if output_ext else filename
+                )
+                return os.path.join(output, output_filename)
+            else:
+                # The output is considered a target file
+                return output
+
+        def iter_paths(source: Sequence, output: Sequence[str]) -> Iterator[Tuple]:
+            if len(output) != 1:
+                for s, o in zip(source, output):
+                    if tiledb.object_type(s) == "group":
+                        logger.debug("Pair %s and %s", s, o)
+                        yield s, create_output_path(s, o)
+                    else:
+                        logger.debug("Input %s is not a tiledb asset", s)
+                        continue
+            else:
+                logger.debug("Traverse source: %s", source)
+                for s in source:
+                    if tiledb.object_type(s) != "group":
+                        # Folder for exploration
+                        contents = vfs.ls(s)
+                        # Explore folders only at depth 1
+                        filtered_contents = [c for c in contents if not vfs.is_dir(c)]
+                        yield from iter_paths(filtered_contents, output)
+                    else:
+                        logger.debug("Pair %s and %s", s, output[0])
+                        yield s, create_output_path(s, output[0])
+
+        logger.debug("Create pairs between %s and %s", source, output)
+        return tuple(iter_paths(source, output))
 
     def build_input_batches(
-        source: Sequence[str], output: str, num_batches: int, out_ext: str
+        source: Sequence[str],
+        output: Sequence[str],
+        num_batches: int,
+        out_ext: str,
+        *,
+        verbose: bool,
     ):
+        logger = get_logger_wrapper(verbose)
+
         """Groups input URIs into batches."""
-        uri_pairs = build_io_uris_exportation(source, output, out_ext)
+        uri_pairs = build_io_uris_exportation(source, output, out_ext, logger)
         # If the user didn't specify a number of batches, run every import
         # as its own task.
+        logger.debug(f"Input batches:{uri_pairs}")
```

Comment:
nit: when logging, use %-style strings and suffixes, so

```python
logger.debug("Input batches: %s", uri_pairs)
```

Reply:
@ktsitsi please use `%r` instead of `%s` so that we get the maximum information (like the difference between `b"1"` and `"1"`).